src/test/resources/processor_test_files/2184_CastArithmeticOperand/NOCOMPILE_BinaryOperatorInMethodCallThatCanReceiveDifferentNumberOfParameters.java

```diff
@@ -0,0 +1,19 @@
+/*
+When a binary operator is in a method call that can receive different number of parameters (e.g., `void info(String var1, Object... var2);`),
+we can get an IndexOutOfBoundsException when trying to find the type expected for the argument (i.e., the binary operator in this case).
+This test checks that such an exception doesn't happen.
```

Comment:
These comments you put in are excellent. We should document in the contributing guidelines that _this_ is how to document a targeted test case like this.